Logger/book.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Logger;
+//We defined each record implicity to make sure data loss is preveneted
+//And so that elements can be accessed without interference 
+public record Book : IEntity
```

Comment:
```suggestion
public record Book : BaseEntity
```
It looks like you did this for your Person record, but make sure you are inheriting from your base entity!

Also, just for consistency, in your Person, Student, and Employee records, you are adding the `class` keyword to the record definition, technically this is done by default. But you should declare your records in a consistent manner and either add that `class` keyword here to be explicit about it or remove it from your other records and let them be class records implicitly.